pkg/kapp/cmd/app/delete.go

```diff
@@ -182,6 +182,14 @@ func (o *DeleteOptions) calculateAndPresentChanges(existingResources []ctlres.Re
 			return ctlcap.ClusterChangeSet{}, nil, false, err
 		}
 
+		// Apply diff-filter on the changes
+		diffFilter, err := o.DiffFlags.DiffFilter()
+		if err != nil {
+			return ctlcap.ClusterChangeSet{}, nil, false, err
+		}
+
+		filteredChanges := diffFilter.Apply(changes)
```

Comment:
nit: `changes = diffFilter.Apply(changes)`. by overriding same var, it's not possible for code below use the wrong var accidently.

Reply:
Yeah sure started overriding same var